week-2-wide-perspective/w02e02-data-manipulation/src/main.ts

```diff
@@ -0,0 +1,28 @@
+import {appendDataToView} from "./orders/tiles/AnnualOrdersTile";
+import  {OrderType} from "./orderType"
+import {highestSoldOrder} from "./orders/tiles/BestSaleTile";
+import {totalOrdesValue} from "./orders/tiles/TotalIncomeTile";
+import {totalOrdersLength} from "./orders/tiles/TotalOrdersTile";
+
+const getOrdersData = async () => {
+    try {
+        const ORDER_URL = `http://127.0.0.1:3000/orders`;
+        const ordersData: OrderType[] = await fetch(ORDER_URL).then((res)=> res.json())
+        return ordersData
+    } catch (error) {
+        console.log("Error",error)
+    }
+
+}
+const showOrdersData = async () => {
+    const ordersData = await getOrdersData()
+    if(!ordersData) return
+    appendDataToView(ordersData)
+    highestSoldOrder(ordersData)
+    totalOrdesValue(ordersData)
+    totalOrdersLength(ordersData.length)
```

Comment:
bardzo sprytnie pomyślne. Wykorzystane dostarczenie danych jako `event` bazowy do rozgłoszenia po wszystkich komponentach :)

To poniżej potraktuj jako "dodatek" - bo jest dobrze, ale chciałbym żebyś zauważył @DragoBJJ to samo co ja:

---

Jest spoko. Postawiłbym tylko kropkę nad `i` - idąc myślą technologiczną rodem z `next.js`.
Bardzo podoba mi się Twoja funkcja:

```javascript
export const appendDataToView = (orderData : OrderType[]) => {}
```
patrząc z perspektywy `AnnualOrderTile.ts`

No i teraz, byłoby bombowo żeby każdy komponent miał właśnie taką, właśnie tak nazwaną.
Co prawda wtedy musielibyśmy tutaj przezywać importy, ale być może nie, jeśli każdy komponent miałby taki kształt:
```typescript
// ...

// Uważaj na odfiltrowanie wg. roku (2022)
const yearToFilter = 2022
subTitle.textContent = `Year ${yearToFilter}`
panel.innerHTML = ''

export const annualOrdersTile = {
  appendDataToView(orderData: OrderType[]) {
    const ordersFilteredByYear = filterOrdersByYear(orderData,2022)
    // Tutaj podobnie, powinniśmy interpretować dane z: ordersFakeData
    for (const { orderDate } of ordersFilteredByYear) {
      panel.appendChild(makeLiElement({ orderDate, orderNumber: '02/2022' }))
    }
  }
}

// Initial render:
annualOrdersTile.appendDataToView([]);

// ...
```

wtedy wykorzystując możliwości `TS` mógłbyś wprowadzić porządek interfacem:
```typescript
interface DataContained<T> {
    appendDataToView(data: T): void
}
```

Finalnie przykładowa implementacja wyglądała by tak:
![image](https://user-images.githubusercontent.com/25398949/199657432-6ee377f3-62b4-4fa4-a906-9599cbb10455.png)